src/main/java/millom/sandbox/mars/weather/service/NasaWeatherService.java

```diff
@@ -1,21 +1,62 @@
 package millom.sandbox.mars.weather.service;
 
 import jakarta.inject.Inject;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
 import millom.sandbox.mars.weather.CustomException.InvalidWeatherException;
+import java.util.Optional;
 import millom.sandbox.mars.weather.mapper.NasaMapper;
-import millom.sandbox.mars.weather.record.NasaWeather;
+import millom.sandbox.mars.weather.pojos.Weather;
+import millom.sandbox.mars.weather.pojos.Sol;
 
 public class NasaWeatherService {
 
   @Inject
   private NasaClientService nasaClientService;
   @Inject
   private NasaMapper nasaMapper;
+  private String dateFormat;
 
-  public NasaWeather getWeatherMapping(String feed, String feedType, float version, String category)
+  public Weather getWeatherMapping(String feed, String feedType, float version, String category)
       throws InvalidWeatherException {
     String marsWeatherAsString = nasaClientService.getJsonWeather(feed, feedType, version,
         category);
     return nasaMapper.deserializeWeather(marsWeatherAsString);
   }
+
+
+  public boolean validateJavaDate(String strDate) {
+    /* Check if date is 'null' */
+    if (strDate.trim().equals("")) {
+      return true;
+    }
+    /* Date is not 'null' */
+    else {
+
+      SimpleDateFormat sdfrmt = new SimpleDateFormat("yyyy-mm-dd");
```

Comment:
the lowercase 'm' in the pattern refers to minutes not month. the uppercase 'M' is intended here.

Reply:
`SimpleDateFormat sdfrmt = new SimpleDateFormat("YYYY-MM-DD");`

Reply:
` else {

       SimpleDateFormat sdfrmt = new SimpleDateFormat("yyyy-MM-dd");`